data-prepper-script/src/main/java/org/opensearch/dataprepper/script/parser/LogListener.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.script.parser;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ErrorNode;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.opensearch.dataprepper.script.antlr.DataPrepperScriptListener;
+import org.opensearch.dataprepper.script.antlr.DataPrepperScriptParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @since 1.3
+ * LogListener is a utility listener that logs every event enter and exit. Useful for debugging and developing new
+ * listeners.
+ */
+public class LogListener implements DataPrepperScriptListener {
```

Comment:
This is similar to my other comment on `TestListener` and `ListenerMatcher`, but there is a lot of code here in `LogListener`. How do we know it is actually working as intended without having thorough, isolated tests for it?